### ISSUE TITLE
Android Connection timeout

### ISSUE THREAD

**Issue by Rossella-Mascia-Neosyn** [capacitor-community/bluetooth-le#600]
**Describe the bug**
In my app it often happens that it times out

**To Reproduce**
1. Connect the device to the ble service
2. See error

**Expected behavior**
It should never return connect timeout

**My code**
```
  const scan = async (): Promise<void> => {
    try {
      await BleClient.initialize();
      const isEnable = await checkBluetoothStatus();
      if (isEnable) {
        const device = await BleClient.requestDevice({
          scanMode: ScanMode.SCAN_MODE_LOW_POWER,
          namePrefix: 'SMART_LOCK',
        });
        await connectBt(true, device);
        await BleClient.stopLEScan();
        console.info('stopped scanning');
      }
    } catch (error) {
      console.error(error);
    }
  };

 const connectBt = async (withState: boolean, deviceParam?: BleDevice) => {
    try {
      withState && setIsLoadingBle(true);
      const deviceStored = await getLocker();
      const device: LockerStorage = { macAddress: (deviceParam?.deviceId ?? '') || (deviceStored.macAddress ?? '') };
      if (!device.macAddress) throw new Error('macAddress not found');
      await BleClient.disconnect(device.macAddress);
      await BleClient.connect(device.macAddress);
      const services = await BleClient.getServices(device.macAddress);
      if (services[3]) {
        deviceParam?.deviceId &&
          setLocker({
            macAddress: deviceParam.deviceId,
            service: services[3],
          });
        withState && setIsLoadingBle(false);
        withState && setIsConnectBle(true);
      } else {
        console.error('service not found', services);
      }
    } catch (e) {
      console.error(e);
      setIsLoadingBle(false);
      setIsConnectBle(false);
      await toast({
        behavior: 'error',
        message: (e as ErrorBle).message,
        duration: 5000,
      });
    }
  };
```
**Screenshots Logcat**
![image](https://github.com/capacitor-community/bluetooth-le/assets/79088170/3ca3b321-6132-4f81-9542-1be121295fe4)

**Plugin version:**
- @capacitor-community/bluetooth-le: 3.1.1

**Smartphone (please complete the following information):**
- Device: Samsung Galaxy S8
- Android version 9

**Additional context**
I have already tried, as shown in my code, to insert disconnect and connect together

**Comment by mutasimissa** [comment]
I don’t think this is a bug, the timeout means failed of connecting to a certain deviceId, please try to make sure you are connecting to the correct deviceId and if you are sure then try connecting to other peripheral and see if the problem is the same.

**Comment by Rossella-Mascia-Neosyn** [comment]
actually it's just a bug I tried with another device and I have the same problem.
I tried with: Redmi note 9 e android version 12.

I hope to be able to fix it together as soon as possible because for us this error is blocking

**Comment by madhan-SEET** [comment]
I'm also facing the same issue (Connection Timeout) . I tried to disconnect before connect still Its not working .

**Comment by quadratech-dev-admin** [comment]
Any solution for this?

**Comment by madhan-SEET** [comment]
I just increase my ble device advertisement packets previously 1 sec a packet and now 200 mill sec a packet . Now the connection timeout error decreased . Its not coming frequently but I cant able to avoid totally .

**Comment by peitschie** [comment]
The BLE advertising interval absolutely can impact this significantly. Even 200ms is still only middling. Apple recommends an interval of 20ms for at least the first 30s for best discovery, then has some suggested intervals to slow down at.

See https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf

![image](https://github.com/capacitor-community/bluetooth-le/assets/1052079/ff1fb052-f399-4aa7-b19f-a424bf9d28bc)

**Comment by Cypher103360** [comment]
I have two BLE devices with the same firmware version:

✅ Device 1:
**Connects instantly** — BluetoothGatt callbacks are smooth, with no delay in onConnectionStateChange or onServicesDiscovered.

**Sends data smoothly** — responds correctly to commands like BATTERY, characteristics are read/written successfully, and notifications work as expected.

❌ Device 2:
**Fails to connect reliably** — it either times out or gets stuck before onServicesDiscovered, and sometimes never reaches the connected state.

**Doesn’t respond to commands** — even if the connection is established, it doesn't reply to commands like BATTERY, VER, and characteristic reads return nothing.

So, my question is:

What could be the possible reason? Might there be an issue in the code or something wrong with the BLE device or the firmware version?

**Comment by peitschie** [comment]
Hi @Cypher103360 

Do you see this same situation with another application (e.g., nrf Connect)?

Also, just to confirm, you're experiencing this issue on Android?

**Comment by Cypher103360** [comment]
Hello @peitschie, Thanks a lot for your reply 😊

I tried connecting both devices to the **nRF Connect** application. Eventually, there was no issue with the connection; I was able to connect both devices instantly.

However, when I tried to communicate with the first device with commands like (BATTERY, VER, GETDATA), it replied in  **HEX**, but the second device didn't reply to even a single command.

| Device 1 | Device 2 |
|--------|-------|
| <img src="https://github.com/user-attachments/assets/d2ae2150-872e-41a0-a103-892e0a519c83" width="600">| <img src="https://github.com/user-attachments/assets/b2fe0b29-d894-4b5a-990c-a9f30dc9101a" width="600"> |
